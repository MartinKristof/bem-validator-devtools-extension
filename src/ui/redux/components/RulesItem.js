import React from 'react';
import PropTypes from 'prop-types';

const RulesItem = ({ rule: { missingClassName, selector }, inspectElement }) => (
  <tr className="rules__item">
    <td>
      <strong>{missingClassName}</strong>
    </td>
    <td>
      <a href="#" className="rules__item-link" onClick={() => inspectElement(selector)}>
        <em>{selector}</em>
      </a>
    </td>
  </tr>
);

RulesItem.propTypes = {
  rule: PropTypes.exact({
    missingClassName: PropTypes.string.isRequired,
    selector: PropTypes.string.isRequired,
  }).isRequired,
  inspectElement: PropTypes.func.isRequired,
};

export default RulesItem;
