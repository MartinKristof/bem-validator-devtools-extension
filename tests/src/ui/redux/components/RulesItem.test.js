import React from 'react';
import { shallow } from 'enzyme';
import RulesItem from '../../../../../src/ui/redux/components/RulesItem';

describe('<RulesItem />', () => {
  describe('#render', () => {
    const inspectElement = jest.fn();
    const rule = { missingClassName: '.foo', selector: '.bar' };

    it('should render without side-effects', () => {
      const node = shallow(<RulesItem rule={rule} inspectElement={inspectElement} />);

      expect(node).toMatchSnapshot();
    });

    it('should inspect element', () => {
      const node = shallow(<RulesItem rule={rule} inspectElement={inspectElement} />);

      node.find('a').simulate('click');

      expect(inspectElement).toHaveBeenCalledWith('.bar');
    });
  });
});
