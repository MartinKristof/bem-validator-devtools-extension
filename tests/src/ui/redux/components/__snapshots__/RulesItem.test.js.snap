// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RulesItem /> #render should render without side-effects 1`] = `
<tr>
  <td>
    <strong>
      .foo
    </strong>
  </td>
  <td>
    <a
      href="#"
      onClick={[Function]}
    >
      <em>
        .bar
      </em>
    </a>
  </td>
</tr>
`;
